src/client/src/app/services/socket.service.ts

```diff
@@ -26,6 +26,11 @@ export class SocketService {
 
   createTournament(){
     this.router.navigate(['/tournament/lobby'])
-    this.socket.emit('create-room', this.socket.ioSocket.id)
+    this.socket.emit('create-tournament', this.socket.ioSocket.id)
+    this.socket.on('generate tournament pin', (data :string) => {
```

Comment:
socket has no access to this 'on'.. need to be initialized in the constructor with other on listeners

Reply:
will refactor after pushing your PR